 # Cleaning CTA Data
 
     Charlie Marshall
     Start Date: 16 April 2021
     End Date:

Objective: to clean up data scraped from CTA Train Tracker API to perform data analysis

## Load Libraries

In [68]:
import pandas as pd
import numpy as np
import scipy
import re
import glob
import os
from nltk.tokenize import word_tokenize,sent_tokenize,RegexpTokenizer
from nltk import pos_tag
import spacy

## Load Data

In [469]:
game = pd.read_csv('/Users/charlesmarshall/Desktop/TransMinor/CTA_Project/Purple51421AM.txt', engine='python', sep = ',', names = ['rn','station','isArr','direction','arrtime'])

In [470]:
game_nodup = game.drop_duplicates().reset_index(drop=True)

In [471]:
game = clean_data(game_nodup).reset_index(drop=True)

In [472]:
# game = game_nodup.reset_index(drop=True)

In [473]:
runs = pd.DataFrame(columns=['rn','Central5','Noyes5','Foster5','Davis5','Dempster5','Main5','South Boulevard5','Howard5','Wilson5','Belmont5','Wellington5','Diversey5','Fullerton5','Armitage','Sedgwick5','Chicago5','Merchandise Mart5','Clark/Lake5','State/Lake5','Washington/Wabash5','Adams/Wabash5','Harold Washington Library-State/Van Buren5','LaSalle/Van Buren5','Quincy5','Washington/Wells5','Merchandise Mart1','Chicago1','Sedgewick1','Armitage1','Fullerton1','Diversey1','Wellington1','Belmont1','Wilson1','Howard1','South Boulevard1','Main','Dempster1','Davis1','Foster1','Noyes1','Central1','Linden1'])
# runs = pd.DataFrame(columns=['rn','Central5','South Boulevard5','Howard5','Wilson5','Belmont5','Wellington5','Fullerton5','Armitage','Merchandise Mart5','Clark/Lake5','Adams/Wabash5','Harold Washington Library-State/Van Buren5','Washington/Wells5','Merchandise Mart1','Armitage1','Fullerton1','Diversey1','Wellington1','Belmont1','Wilson1','South Boulevard1','Linden1'])
# runs = pd.DataFrame(columns=['rn','Central5','Davis5','Howard5','Wilson5','Belmont5','Fullerton5','Merchandise Mart5','Adams/Wabash5','Merchandise Mart1','Fullerton1','Belmont1','Wilson1','Howard1','South Boulevard1','Davis1','Linden1'])

In [474]:
runs_sorted = the_whole_deal(game)

In [475]:
runs_sorted

rn  Central5    Noyes5 Foster5    Davis5 Dempster5     Main5  \
0    502  05:05:01      None    None  05:10:01  05:11:26  05:13:03   
1    503  05:20:54      None    None  05:25:58  05:27:25  05:29:00   
2    501  05:38:19      None    None  05:44:36      None  05:47:30   
3    504      None  05:51:08    None  05:55:30      None  05:57:17   
4    506  06:05:13  06:08:12    None  06:11:34      None  06:14:32   
..   ...       ...       ...     ...       ...       ...       ...   
120  515      None      None    None      None      None      None   
121  515      None      None    None      None      None      None   
122  515      None      None    None      None      None      None   
123  515  09:55:50      None    None  10:00:00  10:01:31  10:03:04   
124  516  11:17:58      None    None  11:22:47  11:24:11  11:26:35   

    South Boulevard5   Howard5   Wilson5  ...   Wilson1   Howard1  \
0           05:14:20  05:15:33  05:24:37  ...  06:22:26  06:32:23   
1           05:30:18  05:31:26  05:39:39  ...      None  06:45:13   
2           05:48:19  05:49:04  05:58:31  ...  06:55:31  07:05:05   
3           05:59:08  05:59:47  06:08:31  ...      None  07:12:09   
4           06:15:25  06:16:08  06:25:34  ...  07:26:38  07:36:12   
..               ...       ...       ...  ...       ...       ...   
120             None      None      None  ...      None  09:25:34   
121             None      None      None  ...      None      None   
122             None      None      None  ...      None      None   
123         10:04:22  10:05:37      None  ...      None      None   
124         11:27:28  11:28:08      None  ...      None      None   

    South Boulevard1      Main Dempster1    Davis1   Foster1    Noyes1  \
0               None      None      None      None      None      None   
1           06:48:32      None      None      None      None      None   
2           07:07:01      None      None  05:20:01  05:21:25  05:21:49   
3               None      None      None      None      None      None   
4               None      None      None      None      None      None   
..               ...       ...       ...       ...       ...       ...   
120         09:25:28      None      None      None      None      None   
121             None  09:30:37  09:31:04  09:32:26      None  09:35:24   
122             None      None      None      None      None      None   
123             None      None      None      None      None      None   
124             None      None      None      None      None      None   

     Central1   Linden1  
0        None      None  
1        None      None  
2    05:23:37  05:25:08  
3        None      None  
4        None      None  
..        ...       ...  
120      None      None  
121  09:37:17  09:38:34  
122      None  09:38:52  
123      None      None  
124      None      None  

[125 rows x 44 columns]

In [476]:
runs.to_excel("Excel/Purple51421AM.xlsx")

In [433]:
runs_sorted.to_excel("Excel/Purple51421PM_sorted.xlsx")

In [444]:
clean_data(game[game['rn'] == '521'].reset_index(drop=True)).reset_index(drop=True).head(50)

rn           station isArr direction   arrtime
0   521           Central     1         5  16:18:56
1   521             Davis     0         5  16:20:44
2   521             Davis     1         5  16:23:00
3   521          Dempster     0         5  16:25:23
4   521              Main     1         5  16:25:51
5   521    SouthBoulevard     1         5  16:27:37
6   521            Howard     1         5  16:28:21
7   521            Wilson     0         5  16:36:37
8   521            Wilson     1         5  16:36:39
9   521          Sheridan     0         5  16:39:30
10  521          Sheridan     1         5  16:39:03
11  521           Belmont     0         5  16:43:02
12  521           Belmont     1         5  16:43:35
13  521        Wellington     1         5  16:44:19
14  521          Diversey     1         5  16:46:22
15  521         Fullerton     1         5  16:48:08
16  521          Armitage     0         5  16:50:24
17  521          Armitage     1         5  16:50:33
18  521          Sedgwick     0         5  16:53:36
19  521          Sedgwick     1         5  16:54:19
20  521           Chicago     0         5  16:57:23
21  521           Chicago     1         5  16:57:33
22  521   MerchandiseMart     1         5  17:00:39
23  521        Clark/Lake     0         5  17:02:04
24  521  Washington/Wells     1         5  17:02:27
25  521   MerchandiseMart     0         1  17:04:37
26  521   MerchandiseMart     1         1  17:05:09
27  521           Chicago     0         1  17:07:00
28  521           Chicago     1         1  17:07:32
29  521          Sedgwick     0         1  17:10:24
30  521          Sedgwick     1         1  17:10:35
31  521          Armitage     0         1  17:14:34
32  521          Armitage     1         1  17:14:37
33  521         Fullerton     0         1  17:16:16
34  521         Fullerton     1         1  17:16:31
35  521          Diversey     0         1  17:17:55
36  521          Diversey     1         1  17:18:22
37  521        Wellington     0         1  17:20:09
38  521        Wellington     1         1  17:20:38
39  521            Wilson     1         1  17:27:31
40  521    SouthBoulevard     0         1  17:37:52
41  521            Howard     0         1  17:35:36
42  521              Main     1         1  17:40:19
43  521          Dempster     1         1  17:41:33
44  521             Davis     0         1  17:42:56
45  521            Foster     0         1  17:44:05
46  521             Noyes     1         1  17:44:37
47  521           Central     1         1  17:46:32
48  521            Linden     0         1  17:48:04
49  521            Linden     1         1  17:49:32

In [441]:
runs[runs['rn'] == '521']

Empty DataFrame
Columns: [rn, Central5, Noyes5, Foster5, Davis5, Dempster5, Main5, South Boulevard5, Howard5, Wilson5, Belmont5, Wellington5, Diversey5, Fullerton5, Armitage, Sedgwick5, Chicago5, Merchandise Mart5, Clark/Lake5, State/Lake5, Washington/Wabash5, Adams/Wabash5, Harold Washington Library-State/Van Buren5, LaSalle/Van Buren5, Quincy5, Washington/Wells5, Merchandise Mart1, Chicago1, Sedgewick1, Armitage1, Fullerton1, Diversey1, Wellington1, Belmont1, Wilson1, Howard1, South Boulevard1, Main, Dempster1, Davis1, Foster1, Noyes1, Central1, Linden1]
Index: []

[0 rows x 44 columns]

In [138]:
runs.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 52 entries, 0 to 51
Data columns (total 44 columns):
 #   Column                                      Non-Null Count  Dtype 
---  ------                                      --------------  ----- 
 0   rn                                          52 non-null     object
 1   Central5                                    36 non-null     object
 2   Noyes5                                      0 non-null      object
 3   Foster5                                     0 non-null      object
 4   Davis5                                      37 non-null     object
 5   Dempster5                                   20 non-null     object
 6   Main5                                       36 non-null     object
 7   South Boulevard5                            37 non-null     object
 8   Howard5                                     38 non-null     object
 9   Wilson5                                     24 non-null     object
 10  Belmont5                    

In [345]:
# For when "isArr" is added, that needs to be part of the criteria also

def clean_data(game):
    games = []
    for i in range(len(game)):
        if i == 0:
            game.loc[i][0]=re.sub(r"'",'',game.loc[i][0])
            game.loc[i][0]=re.sub(r'\[','',game.loc[i][0])

            game.loc[i][1]=re.sub(r"'",'',game.loc[i][1])
            game.loc[i][1]=re.sub(r' ','',game.loc[i][1])
            game.loc[i][2]=re.sub(r"'",'',game.loc[i][2])
            game.loc[i][2]=re.sub(r' ','',game.loc[i][2])
            game.loc[i][3]=re.sub(r"'",'',game.loc[i][3])
            game.loc[i][3]=re.sub(r' ','',game.loc[i][3])

            game.loc[i][4]=re.sub(r"'",'',game.loc[i][4])
            game.loc[i][4]=re.sub(r'\]','',game.loc[i][4])
            game.loc[i][4]=re.sub(r' 2021[0-9]{4} ','',game.loc[i][4])
        elif game.loc[i].rn == game.loc[i-1].rn and game.loc[i].station == game.loc[i-1].station and game.loc[i].isArr == game.loc[i-1].isArr and game.loc[i].direction == game.loc[i-1].direction:
            #print(i-1)
            games.append(i-1)
        
        game.loc[i][0]=re.sub(r"'",'',game.loc[i][0])
        game.loc[i][0]=re.sub(r'\[','',game.loc[i][0])
        
        game.loc[i][1]=re.sub(r"'",'',game.loc[i][1])
        game.loc[i][1]=re.sub(r' ','',game.loc[i][1])
        game.loc[i][2]=re.sub(r"'",'',game.loc[i][2])
        game.loc[i][2]=re.sub(r' ','',game.loc[i][2])
        game.loc[i][3]=re.sub(r"'",'',game.loc[i][3])
        game.loc[i][3]=re.sub(r' ','',game.loc[i][3])
        
        game.loc[i][4]=re.sub(r"'",'',game.loc[i][4])
        game.loc[i][4]=re.sub(r'\]','',game.loc[i][4])
        game.loc[i][4]=re.sub(r' 2021[0-9]{4} ','',game.loc[i][4])
        
        
    game = game.drop(games)
            
    return game

In [346]:
game.rn.unique()

array(['514', '517', '518', '513', '515', '520', '519', '506', '508',
       '507', '522', '509', '523', '521', '512', '516', '1001', '1000'],
      dtype=object)

In [347]:
runs.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 53 entries, 0 to 52
Data columns (total 44 columns):
 #   Column                                      Non-Null Count  Dtype 
---  ------                                      --------------  ----- 
 0   rn                                          53 non-null     object
 1   Central5                                    35 non-null     object
 2   Noyes5                                      1 non-null      object
 3   Foster5                                     0 non-null      object
 4   Davis5                                      35 non-null     object
 5   Dempster5                                   23 non-null     object
 6   Main5                                       34 non-null     object
 7   South Boulevard5                            34 non-null     object
 8   Howard5                                     35 non-null     object
 9   Wilson5                                     22 non-null     object
 10  Belmont5                    

In [371]:
def run_funct(df_rn):

# Problems:
# This prolly wont work with all run numbers
# FIXED: RN 515 doesnt work because the second loop run didn't start with Central (Davis)
# FIXED: RN 514 doesnt split up any of the runs at all (loop and non-loop)
# FIXED: RN 517 doesnt split up non-loop runs

# Changes:
# This needs to only work with loop runs

# This function probably could use some work

    runs = []
    first = 0
#     second = None

    for i, row in df_rn.iterrows():
        # Dealing with the end of the data: If we are at the end of the dataset, the "last run" will be from its 
        # begining to the last point recorded (this might not be the end of a run in the case of a time cutoff)
        if i == len(df_rn)-1:
            run=[]
            run.append(row.rn)
            for j in range(first,len(df_rn)):
                run.append((df_rn.loc[j].station,df_rn.loc[j].direction,df_rn.loc[j].arrtime,df_rn.loc[j].isArr))
            runs.append(run)
        # If we have a purple line, non-loop run SB from Linden to Howard
        elif df_rn.loc[i].station == 'Howard' and df_rn.loc[i+1].station != 'Wilson' :
            run = []
            run.append(row.rn)
            for j in range(first,i+1):
                run.append((df_rn.loc[j].station,df_rn.loc[j].direction,df_rn.loc[j].arrtime,df_rn.loc[j].isArr))
            runs.append(run)
            first = i+1
        #If we have a NB purple shuttle (Howard to Linden)
        elif df_rn.loc[i].station == 'Linden' and df_rn.loc[first].station != 'Howard' :
            run = []
            run.append(row.rn)
            for j in range(first,i+1):
                run.append((df_rn.loc[j].station,df_rn.loc[j].direction,df_rn.loc[j].arrtime,df_rn.loc[j].isArr))
            runs.append(run)
            first = i+1
        # If we have a purple line loop run
        elif df_rn.loc[i].station == 'Linden' and df_rn.loc[first].station == 'Central':
            run = []
            run.append(row.rn)
            for j in range(first,i+1):
                run.append((df_rn.loc[j].station,df_rn.loc[j].direction,df_rn.loc[j].arrtime,df_rn.loc[j].isArr))
            runs.append(run)
            first = i+1
            
    final_runs = []
    for i, train in enumerate(runs):
        if len(train) > 10:
            final_runs.append(train)
#             print(train)

#     return final_runs
    return runs

In [361]:
def put_it_in_df(lists):
    
    for i in range(len(lists)):
        run = [None]*len(runs.columns)
        run[0] = lists[i][0]
        for j in range(len(lists[i])):
            if lists[i][j][0] == 'Central' and lists[i][j][1] == '5':
                run[1] = lists[i][j][2]
            if lists[i][j][0] == 'Noyes' and lists[i][j][1] == '5':
                run[2] = lists[i][j][2]
            if lists[i][j][0] == 'Foster' and lists[i][j][1] == '5':
                run[3] = lists[i][j][2]
            if lists[i][j][0] == 'Davis' and lists[i][j][1] == '5':
                run[4] = lists[i][j][2]
            if lists[i][j][0] == 'Dempster' and lists[i][j][1] == '5':
                run[5] = lists[i][j][2]
            if lists[i][j][0] == 'Main' and lists[i][j][1] == '5':
                run[6] = lists[i][j][2]
            if lists[i][j][0] == 'SouthBoulevard' and lists[i][j][1] == '5':
                run[7] = lists[i][j][2]
            if lists[i][j][0] == 'Howard' and lists[i][j][1] == '5':
                run[8] = lists[i][j][2]
            if lists[i][j][0] == 'Wilson' and lists[i][j][1] == '5':
                run[9] = lists[i][j][2]
            if lists[i][j][0] == 'Belmont' and lists[i][j][1] == '5':
                run[10] = lists[i][j][2]
            if lists[i][j][0] == 'Wellington' and lists[i][j][1] == '5':
                run[11] = lists[i][j][2]
            if lists[i][j][0] == 'Diversey' and lists[i][j][1] == '5':
                run[12] = lists[i][j][2]
            if lists[i][j][0] == 'Fullerton' and lists[i][j][1] == '5':
                run[13] = lists[i][j][2]
            if lists[i][j][0] == 'Armitage' and lists[i][j][1] == '5':
                run[14] = lists[i][j][2]
            if lists[i][j][0] == 'Sedgwick' and lists[i][j][1] == '5':
                run[15] = lists[i][j][2]
            if lists[i][j][0] == 'Chicago' and lists[i][j][1] == '5':
                run[16] = lists[i][j][2]
            if lists[i][j][0] == 'MerchandiseMart' and lists[i][j][1] == '5':
                run[17] = lists[i][j][2]
            if lists[i][j][0] == 'Clark/Lake' and lists[i][j][1] == '5':
                run[18] = lists[i][j][2]
            if lists[i][j][0] == 'State/Lake' and lists[i][j][1] == '5':
                run[19] = lists[i][j][2]
            if lists[i][j][0] == 'Washington/Wabash' and lists[i][j][1] == '5':
                run[20] = lists[i][j][2]
            if lists[i][j][0] == 'Adams/Wabash' and lists[i][j][1] == '5':
                run[21] = lists[i][j][2]
            if lists[i][j][0] == 'HaroldWashingtonLibrary-State/VanBuren' and lists[i][j][1] == '5':
                run[22] = lists[i][j][2]
            if lists[i][j][0] == 'LaSalle/VanBuren' and lists[i][j][1] == '5':
                run[23] = lists[i][j][2]
            if lists[i][j][0] == 'Quincy' and lists[i][j][1] == '5':
                run[24] = lists[i][j][2]
            if lists[i][j][0] == 'Washington/Wells' and lists[i][j][1] == '5':
                run[25] = lists[i][j][2]
            if lists[i][j][0] == 'MerchandiseMart' and lists[i][j][1] == '1':
                run[26] = lists[i][j][2]
            if lists[i][j][0] == 'Chicago' and lists[i][j][1] == '1':
                run[27] = lists[i][j][2]
            if lists[i][j][0] == 'Sedgwick' and lists[i][j][1] == '1':
                run[28] = lists[i][j][2]
            if lists[i][j][0] == 'Armitage' and lists[i][j][1] == '1':
                run[29] = lists[i][j][2]
            if lists[i][j][0] == 'Fullerton' and lists[i][j][1] == '1':
                run[30] = lists[i][j][2]
            if lists[i][j][0] == 'Diversey' and lists[i][j][1] == '1':
                run[31] = lists[i][j][2]
            if lists[i][j][0] == 'Wellington' and lists[i][j][1] == '1':
                run[32] = lists[i][j][2]
            if lists[i][j][0] == 'Belmont' and lists[i][j][1] == '1':
                run[33] = lists[i][j][2]
            if lists[i][j][0] == 'Wilson' and lists[i][j][1] == '1':
                run[34] = lists[i][j][2]
            if lists[i][j][0] == 'Howard' and lists[i][j][1] == '1':
                run[35] = lists[i][j][2]
            if lists[i][j][0] == 'SouthBoulevard' and lists[i][j][1] == '1':
                run[36] = lists[i][j][2]
            if lists[i][j][0] == 'Main' and lists[i][j][1] == '1':
                run[37] = lists[i][j][2]
            if lists[i][j][0] == 'Dempster' and lists[i][j][1] == '1':
                run[38] = lists[i][j][2]
            if lists[i][j][0] == 'Davis' and lists[i][j][1] == '1':
                run[39] = lists[i][j][2]
            if lists[i][j][0] == 'Foster' and lists[i][j][1] == '1':
                run[40] = lists[i][j][2]
            if lists[i][j][0] == 'Noyes' and lists[i][j][1] == '1':
                run[41] = lists[i][j][2]
            if lists[i][j][0] == 'Central' and lists[i][j][1] == '1':
                run[42] = lists[i][j][2]
            if lists[i][j][0] == 'Linden' and lists[i][j][1] == '1':
                run[43] = lists[i][j][2]

    #     print(run)
        runs.loc[len(runs)] = run

    return runs

In [141]:
# def put_it_in_df(lists):
    
#     for i in range(len(lists)):
#         run = [None]*len(runs.columns)
#         run[0] = lists[i][0]
#         for j in range(len(lists[i])):
#             if lists[i][j][0] == 'Central' and lists[i][j][1] == '5':
#                 run[1] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Noyes'" and lists[i][j][1] == " '5'":
# #                 run[2] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Foster'" and lists[i][j][1] == " '5'":
# #                 run[3] = lists[i][j][2]
#             if lists[i][j][0] == 'Davis' and lists[i][j][1] == '5':
#                 run[2] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Dempster'" and lists[i][j][1] == " '5'":
# #                 run[5] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Main'" and lists[i][j][1] == " '5'":
# #                 run[6] = lists[i][j][2]
# #             if lists[i][j][0] == " 'South Boulevard'" and lists[i][j][1] == " '5'":
# #                 run[2] = lists[i][j][2]
#             if lists[i][j][0] == 'Howard' and lists[i][j][1] == '5':
#                 run[3] = lists[i][j][2]
#             if lists[i][j][0] == 'Wilson' and lists[i][j][1] == '5':
#                 run[4] = lists[i][j][2]
#             if lists[i][j][0] == 'Belmont' and lists[i][j][1] == '5':
#                 run[5] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Wellington'" and lists[i][j][1] == " '5'":
# #                 run[6] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Diversy'" and lists[i][j][1] == " '5'":
# #                 run[7] = lists[i][j][2]
#             if lists[i][j][0] == 'Fullerton' and lists[i][j][1] == '5':
#                 run[6] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Armitage'" and lists[i][j][1] == " '5'":
# #                 run[8] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Sedgwick'" and lists[i][j][1] == " '5'":
# #                 run[15] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Chicago'" and lists[i][j][1] == " '5'":
# #                 run[16] = lists[i][j][2]
#             if lists[i][j][0] == 'MerchandiseMart' and lists[i][j][1] == '5':
#                 run[7] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Clark/Lake'" and lists[i][j][1] == " '5'":
# #                 run[10] = lists[i][j][2]
# #             if lists[i][j][0] == " 'State/Lake'" and lists[i][j][1] == " '5'":
# #                 run[19] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Washington/Wabash'" and lists[i][j][1] == " '5'":
# #                 run[20] = lists[i][j][2]
#             if lists[i][j][0] == 'Adams/Wabash' and lists[i][j][1] == '5':
#                 run[8] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Harold Washington Library-State/Van Buren'" and lists[i][j][1] == " '5'":
# #                 run[12] = lists[i][j][2]
# #             if lists[i][j][0] == " 'LaSalle/Van Buren'" and lists[i][j][1] == " '5'":
# #                 run[23] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Quincy'" and lists[i][j][1] == " '5'":
# #                 run[24] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Washington/Wells'" and lists[i][j][1] == " '5'":
# #                 run[13] = lists[i][j][2]
#             if lists[i][j][0] == 'MerchandiseMart' and lists[i][j][1] == '1':
#                 run[9] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Chicago'" and lists[i][j][1] == " '1'":
# #                 run[27] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Sedgwick'" and lists[i][j][1] == " '1'":
# #                 run[28] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Armitage'" and lists[i][j][1] == " '1'":
# #                 run[15] = lists[i][j][2]
#             if lists[i][j][0] == 'Fullerton' and lists[i][j][1] == '1':
#                 run[10] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Diversey'" and lists[i][j][1] == " '1'":
# #                 run[17] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Wellington'" and lists[i][j][1] == " '1'":
# #                 run[18] = lists[i][j][2]
#             if lists[i][j][0] == 'Belmont' and lists[i][j][1] == '1':
#                 run[11] = lists[i][j][2]
#             if lists[i][j][0] == 'Wilson' and lists[i][j][1] == '1':
#                 run[12] = lists[i][j][2]
#             if lists[i][j][0] == " 'Howard'" and lists[i][j][1] == " '1'":
#                 run[13] = lists[i][j][2]
#             if lists[i][j][0] == 'SouthBoulevard' and lists[i][j][1] == '1':
#                 run[14] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Main'" and lists[i][j][1] == " '1'":
# #                 run[37] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Dempster'" and lists[i][j][1] == " '1'":
# #                 run[38] = lists[i][j][2]
#             if lists[i][j][0] == 'Davis' and lists[i][j][1] == '1':
#                 run[15] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Foster'" and lists[i][j][1] == " '1'":
# #                 run[40] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Noyes'" and lists[i][j][1] == " '1'":
# #                 run[41] = lists[i][j][2]
# #             if lists[i][j][0] == " 'Central'" and lists[i][j][1] == " '1'":
# #                 run[42] = lists[i][j][2]
#             if lists[i][j][0] == 'Linden' and lists[i][j][1] == '1':
#                 run[16] = lists[i][j][2]

#     #     print(run)
#         runs.loc[len(runs)] = run

#     return runs

In [362]:
def the_whole_deal(game):

    for i, r in enumerate(game.rn.unique()):

        gm = clean_data(game[game.rn == r].reset_index(drop=True)).reset_index(drop=True)
        gmr = run_funct(gm)
        runsdf = put_it_in_df(gmr)
        
    return runsdf.sort_values(by =['Merchandise Mart5']).reset_index(drop=True)

In [143]:
runs.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 52 entries, 0 to 51
Data columns (total 44 columns):
 #   Column                                      Non-Null Count  Dtype 
---  ------                                      --------------  ----- 
 0   rn                                          52 non-null     object
 1   Central5                                    36 non-null     object
 2   Noyes5                                      0 non-null      object
 3   Foster5                                     0 non-null      object
 4   Davis5                                      37 non-null     object
 5   Dempster5                                   20 non-null     object
 6   Main5                                       36 non-null     object
 7   South Boulevard5                            37 non-null     object
 8   Howard5                                     38 non-null     object
 9   Wilson5                                     24 non-null     object
 10  Belmont5                    

rn506 = clean_data(game[game.rn == "'506'"].reset_index(drop=True)).reset_index(drop=True)

rn506

runs516 = run_funct(rn516)

put_it_in_df(runs515)

put_it_in_df(runs516)

rn516 = clean_data(game[game.rn == "'516'"].reset_index(drop=True)).reset_index(drop=True)

rn516

run_funct(rn522)

rn515 = clean_data(game[game.rn == "'515'"].reset_index(drop=True)).reset_index(drop=True)

runs515 = run_funct(rn515)

runs515

rn514 = clean_data(game[game.rn == "'514'"].reset_index(drop=True)).reset_index(drop=True)

rn514

rn517 = clean_data(game[game.rn == "'517'"].reset_index(drop=True)).reset_index(drop=True)

rn517

rn512 = clean_data(game[game.rn == "'512'"].reset_index(drop=True)).reset_index(drop=True)

rn512

rn518 = clean_data(game[game.rn == "'518'"].reset_index(drop=True)).reset_index(drop=True)

rn518

rn504 = clean_data(game[game.rn == "'504'"].reset_index(drop=True)).reset_index(drop=True)

rn504.head()

run_funct(rn518)

rn521 = clean_data(game[game.rn == "'521'"].reset_index(drop=True)).reset_index(drop=True)

rn521.head()

rn508 = clean_data(game[game.rn == "'508'"].reset_index(drop=True)).reset_index(drop=True)

rn508.head()

rn509 = clean_data(game[game.rn == "'509'"].reset_index(drop=True)).reset_index(drop=True)

rn509.head()

rn519 = clean_data(game[game.rn == "'519'"].reset_index(drop=True)).reset_index(drop=True)

rn519

rn521 = clean_data(game[game.rn == "'521'"].reset_index(drop=True)).reset_index(drop=True)

rn521.head()

rn522 = clean_data(game[game.rn == "'522'"].reset_index(drop=True)).reset_index(drop=True)

rn522

rn1000 = clean_data(game[game.rn == "'1000'"].reset_index(drop=True)).reset_index(drop=True)

rn1000

runs1000 = run_funct(rn1000)

put_it_in_df(runs1000)